numba/cuda/testing.py

```diff
@@ -23,6 +23,28 @@ class CUDATestCase(SerialMixin, TestCase):
     its tests are run between tests from a CUDATestCase.
     """
 
+    @classmethod
+    def setUpClass(cls):
+        # If the test class defines a setUp method, we need to run both
+        # CUDATestCase.setUp (to disable low occupancy warnings) and its setUp
+        if cls is not CUDATestCase and cls.setUp is not CUDATestCase.setUp:
+            cls_setUp = cls.setUp
+
+            def setUp_both(self, *args, **kwargs):
+                # Call our setUp first so that low occupancy warnings are
+                # disabled before anything else happens
+                CUDATestCase.setUp(self)
+                return cls_setUp(self, *args, **kwargs)
+            cls.setUp = setUp_both
```

Comment:
Why is this needed? 

If we need to guarantee that this `setUp` is called first, it's better for all the subclasses to have:

```python
def setUp(self):
    super().setUp()
    .... # local setup
```

Reply:
@sklam is your suggestion that the setupClass method is not required and that a call to the base class setup is recommended as shown?

Reply:
Yes, give that a try.  See if avoiding the `setUpClass` fixes the test failure.